x-pack/examples/third_party_lens_navigation_prompt/public/plugin.ts

```diff
@@ -136,33 +131,53 @@ export class EmbeddedLensExamplePlugin
       ],
     });
 
-    if (!discover.locator) return;
-    lens.registerTopNavMenuEntryGenerator(({ datasourceStates, query, filters }) => {
-      if (!datasourceStates.indexpattern.state) return;
+    lens.registerTopNavMenuEntryGenerator(
+      ({ visualizationId, visualizationState, datasourceStates }) => {
+        if (!datasourceStates.indexpattern.state || !visualizationState) return;
 
-      return {
-        label: 'Go to discover',
-        iconType: 'discoverApp',
-        run: async () => {
-          const [, { data }] = await core.getStartServices();
-          const firstLayer = Object.values(
-            (datasourceStates.indexpattern.state as IndexPatternPersistedState).layers
-          )[0] as PersistedIndexPatternLayer & { indexPatternId: string };
-          discover.locator!.navigate({
-            indexPatternId: firstLayer.indexPatternId,
-            timeRange: data.query.timefilter.timefilter.getTime(),
-            filters,
-            query,
-            columns: firstLayer.columnOrder
-              .map((columnId) => {
-                const column = firstLayer.columns[columnId];
-                if ('sourceField' in column) return column.sourceField;
-              })
-              .filter(Boolean) as string[],
-          });
-        },
-      };
-    });
+        return {
+          label: 'Debug in Playground',
+          iconType: 'wrench',
+          run: async () => {
+            const [coreStart] = await core.getStartServices();
+            const datasourceState = datasourceStates.indexpattern
+              .state as IndexPatternPersistedState;
+            const layersIds = Object.keys(datasourceState.layers);
+            const layers = Object.values(datasourceState.layers) as Array<
+              PersistedIndexPatternLayer & { indexPatternId: string }
+            >;
+            coreStart.application.navigateToApp('testing_embedded_lens', {
+              state: {
+                visualizationType: visualizationId,
+                title: 'Lens visualization',
+                references: [
+                  {
+                    id: layers[0].indexPatternId,
+                    name: 'indexpattern-datasource-current-indexpattern',
+                    type: 'index-pattern',
+                  },
+                  ...layers.map(({ indexPatternId }, i) => ({
+                    id: indexPatternId,
+                    name: `indexpattern-datasource-layer-${layersIds[i]}`,
+                    type: 'index-pattern',
+                  })),
+                ],
+                state: {
+                  datasourceStates: {
+                    indexpattern: {
+                      layers: datasourceState.layers,
+                    },
+                  },
+                  visualization: visualizationState,
+                  filters: [],
```

Comment:
Nit: Any reason we are not carrying over filters and query here?

Reply:
Good point. Effectively those are not useful unless the user clicks again on the `Open in Lens` in the playground. Will update it with them.

Reply:
Why are they not useful? They will become part of the state and affect the shown data, right?

Reply:
Right. I wasn't forwarding them in the playground, but will fix it now.